bin/desi_assemble_tilepix

```diff
@@ -0,0 +1,166 @@
+
+#- read exposures and frames in this production
+log.info(f'Reading exposures from {args.expfile}')
+exp = Table.read(args.expfile, 'EXPOSURES')
+frames = Table.read(args.expfile, 'FRAMES')
+
+#- Load all individual tilepix files
+tileids, firstindex = np.unique(exp['TILEID'], return_index=True) 
+tilepix = dict()
+for tileid, i in zip(tileids, firstindex):
+    night = exp['NIGHT'][i]
+    expid = exp['EXPID'][i]
+    fn = desispec.io.findfile('tilepix', tile=tileid, night=night, expid=expid)
+    with open(fn) as fx:
+        tilepix.update( json.load(fx) )
+
+#- filter out petals that don't have frames
+for tileid in tileids:
+    ii = frames['TILEID'] == tileid
+    camword = parse_cameras(np.unique(frames['CAMERA'][ii]), loglevel='warning')
+    #- should only be complete petals, not individual cameras left
+    assert 'b' not in camword
+    assert 'r' not in camword
+    assert 'z' not in camword
+    for petal in range(10):
+        if camword_union([camword, f'a{petal}']) != camword:
+            log.debug(f'Dropping tileid {tileid} petal {petal} not in camword {camword}')
+            del tilepix[str(tileid)][str(petal)]
+
+if args.outfile:
+    #- convert to a table
+    rows = list()
+    for tileid in tilepix:
+        ii = exp['TILEID'] == int(tileid)
+        survey = exp['SURVEY'][ii][0]
+        program = exp['PROGRAM'][ii][0]
+        for petal in tilepix[tileid]:
+            for pix in tilepix[tileid][petal]:
+                rows.append((int(tileid), survey, program, int(petal),int(pix)))
+
+    tx = Table(rows=rows,
+            names=('TILEID', 'SURVEY', 'PROGRAM', 'PETAL_LOC', 'HEALPIX'),
+            dtype=(np.int32, str, str, np.int16, np.int32) )
+
+    tx.meta['EXTNAME'] = 'TILEPIX'
+    tx.meta['HPXNSIDE'] = 64
+    tx.meta['HPXNEST'] = True
+    tx.write(args.outfile, overwrite=True)
+    log.info(f'Wrote {args.outfile}')
+
+    jsonout = os.path.splitext(args.outfile)[0] + '.json'
+    with open(jsonout, 'w') as fx:
+        json.dump(tilepix, fx)
+    log.info(f'Wrote {jsonout}')
+
+#- if not checking, we're done and can exit
+if not args.check:
+    sys.exit(0)
+
+log.info('Checking for missing or extra healpix')
+nbad = 0
+for survey, program in list(np.unique(exp['SURVEY', 'PROGRAM'])):
+    if isinstance(survey, bytes):
+        survey = survey.decode()
+    if isinstance(program, bytes):
+        program = program.decode()
+
+    ii = (exp['SURVEY'] == survey) & (exp['PROGRAM'] == program)
+    tileids = np.unique(exp['TILEID'][ii])
+    healpix = getpix(tilepix, tileids)
+    log.info(f'{survey} {program} {len(healpix)} healpix')
+    for pix in healpix:
+        fn = desispec.io.findfile('redrock', groupname=pix,
+                survey=survey, faprogram=program)
+        if not os.path.exists(fn):
+            log.error(f'MISSING {fn}')
+            nbad += 1
+
+    #- Check for extra directories
+    for dirname in sorted(glob.glob(f'healpix/{survey}/{program}/*/*')):
```

Comment:
Doesn't glob allow you to specify numeric dir names such that you are less likely to have ValueErrors?

Reply:
that reduces the chance of picking up a non-int dir, but it can still trip on pathological cases like "123blat", so I still have to use the try/except protection when casting to an int.  What I'd really want is something regex-like `[0-9]+` to match 1 or more digits but only digits, but the glob syntax doesn't seem to support that.  I went ahead and added the `[0-9]*` to the code for possible clarity, while keeping the try/except too.